### PR TITLE
Add an endpoint to scan a file

### PR DESCRIPTION
**Pull request by sgarouachi**
- Enhancing the wiki page would be tackled in a separate PR to document changes done for `get_discoveries` and the new endpoint added in this PR 
- Added functional tests

### CODE REVIEW

File: ui/server.py [hunk 1]
```diff
@@ -538,6 +538,63 @@ def update_similar_discoveries():
...
+    if rules_to_use != 'all':
+        app.logger.debug(f'Use rules only from {rules_to_use} category')
+    else:
+        rules_to_use = None
+
+    # Scan
+    try:
+        discoveries = c.scan_path(scan_path=file_path, models=models, force=force_scan, 
+                                  similarity=False, max_depth=-1, ignore_list=[], category=rules_to_use)
+    except Exception as ex:
```
Comment (marcorosa) [inline ui/server.py:579]: as before, better `OSError` here

File: ui/server.py [hunk 2]
```diff
@@ -538,6 +538,63 @@ def update_similar_discoveries():
...
+            f'Error occured when scanning file={filename}, file path={file_path}, error={ex}')
+        os.remove(file_path)
+        return f'Error in scanning file {filename}', 500
+
+    # Get discoveries
+    discoveriesWithRules = []
+    if len(discoveries):
+        try:
+            discoveriesWithRules = c.get_discoveries_with_rules(repo_url=file_path)
+        except Exception as ex:
```
Comment (marcorosa) [inline ui/server.py:590]: again exception type

File: ui/server.py [hunk 3]
```diff
@@ -538,6 +538,63 @@ def update_similar_discoveries():
...
+        discoveries = c.scan_path(scan_path=file_path, models=models, force=force_scan, 
+                                  similarity=False, max_depth=-1, ignore_list=[], category=rules_to_use)
+    except Exception as ex:
+        app.logger.error(
+            f'Error occured when scanning file={filename}, file path={file_path}, error={ex}')
+        os.remove(file_path)
+        return f'Error in scanning file {filename}', 500
+
+    # Get discoveries
+    discoveriesWithRules = []
```
Comment (marcorosa) [inline ui/server.py:586]: please use snake_case for variable names. Not an error, just for pep8 compliance

File: tests/functional_tests/test_get_discoveries_postgres.py [hunk 4]
```diff
@@ -58,7 +58,11 @@ def tearDownClass(cls):
         """ Remove the repo and all its discoveries. """
         cls.client.delete_repo(REPO_URL)
         cls.client.delete_discoveries(REPO_URL)
-        os.remove(cls.csv_path)
+        try:
+            os.remove(cls.csv_path)
+        except Exception as ex:
+            print(f'Failed to cleanup {cls.csv_path}, error={ex}')
+            pass
```
[the remark this answers is not in the document]
Comment (sgarouachi) [reply tests/functional_tests/test_get_discoveries_postgres.py:65]: I'll

File: tests/functional_tests/test_get_discoveries_postgres.py [hunk 5]
```diff
@@ -58,7 +58,11 @@ def tearDownClass(cls):
         """ Remove the repo and all its discoveries. """
         cls.client.delete_repo(REPO_URL)
         cls.client.delete_discoveries(REPO_URL)
-        os.remove(cls.csv_path)
+        try:
+            os.remove(cls.csv_path)
+        except Exception as ex:
```
[the remark this answers is not in the document]
Comment (sgarouachi) [reply tests/functional_tests/test_get_discoveries_postgres.py:63]: I'll

File: ui/backend/client_ui.py [hunk 6]
```diff
@@ -9,6 +9,8 @@
 from git import GitCommandError, InvalidGitRepositoryError, NoSuchPathError
 from git import Repo as GitRepo
```
[the remark this answers is not in the document]
Comment (sgarouachi) [reply ui/backend/client_ui.py:11]: I'll

File: ui/server.py [hunk 7]
```diff
@@ -538,6 +538,63 @@ def update_similar_discoveries():
         return 'OK', 200
 
 
+@app.route('/scan_file', methods=['POST'])
+def scan_file():
```
[the remark this answers is not in the document]
Comment (sgarouachi) [reply ui/server.py:542]: This endpoint is intended to take a file as a parameter and scan it.
When we run the extension locally we grab the file and transferred to be scanned

**Comment by sgarouachi** [comment]
> Thanks @sgarouachi for your contribution. I have some minor requests

Addressed :)

**Review by marcorosa — APPROVED** [review]

**Pull request merged**